### PR TITLE
Adss implementations with pre minting strategy and other improvements

### PR DESCRIPTION
**Pull request by steven2308**
On pre minting implementations, only owner can mint and there's no cost.
Moves IRMRKInitData import to abstract on implementations.
Removes redundant modifier and improves coverage.

### CODE REVIEW

File: test/implementations/premint.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  singleFixtureWithArgs,
+} from '../utils';
+import { Contract } from 'ethers';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { ERC20Mock } from '../../typechain-types';
+
+async function multiAssetFixture(): Promise<Contract> {
+  return await singleFixtureWithArgs('RMRKMultiAssetImplPreMint', [
+    'MultiAsset',
+    'MR',
```
Comment (ThunderDeliverer) [inline test/implementations/premint.ts:18]: Like I mentioned before. It would be cool to rename this to `MA`, for someone taking a look at our codebase, who is unfamiliar with the MultiResource, this could be confusing 🤔 

```suggestion
    'MA',
```

Comment (ThunderDeliverer) [reply test/implementations/premint.ts:18]: I'd suggest doing this for all instances.

File: contracts/implementations/RMRKNestableImpl.sol [hunk 2]
```diff
@@ -54,7 +53,7 @@ contract RMRKNestableImpl is IRMRKInitData, RMRKAbstractNestableImpl {
     function mint(
         address to,
         uint256 numToMint
-    ) public payable virtual notLocked saleIsOpen {
```
Comment (ThunderDeliverer) [inline contracts/implementations/RMRKNestableImpl.sol:57]: Why is this removed? To be less opinionated or just to preserve contract size?

**Review by ThunderDeliverer — APPROVED** [review]
Am I correct to assume that minting in `erc20pay` is not `payable` any more because ERC20 tokens are meant to be used to pay for minting?

If all of the questions that I had were done intentionally, I think this is good to be merged 👌

File: test/implementations/premint.ts (hunk 1 above)
Comment (steven2308) [reply test/implementations/premint.ts:18]: Will do on dev, thanks!

File: contracts/implementations/RMRKNestableImpl.sol (hunk 2 above)
Comment (steven2308) [reply contracts/implementations/RMRKNestableImpl.sol:57]: `saleIsOpen` was checking that  `totaSupply <= maxSupply`. I noticed that the modifier was never used and when tracking it it happens that on minting we check that the current supply + the total being minted doesn't go over the max supply.  So the check on `saleIsOpen` is non sense, it will never be reached.

**Comment by steven2308** [comment]
> Am I correct to assume that minting in `erc20pay` is not `payable` any more because ERC20 tokens are meant to be used to pay for minting?
> 
> If all of the questions that I had were done intentionally, I think this is good to be merged ok_hand

Yep, there was one of the implementations where I had forgotten to remove the payable which is not necessary.
